src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -208,59 +208,57 @@ bool Terminal::SendKeyEvent(const WORD vkey, const DWORD modifiers)
         _NotifyScrollEvent();
     }
 
-    KeyEvent keyEv{ true, 0, vkey, 0, UNICODE_NULL, modifiers };
-
     // AltGr key combinations don't always contain any meaningful,
     // pretranslated unicode character during WM_KEYDOWN.
     // E.g. on a German keyboard AltGr+Q should result in a "@" character,
     // but actually results in "Q" with Alt and Ctrl modifier states.
     // By returning false though, we can abort handling this WM_KEYDOWN
     // event and let the WM_CHAR handler kick in, which will be
     // provided with an appropriate unicode character.
-    if (keyEv.IsAltGrPressed())
+    if (states.IsAltGrPressed())
     {
         return false;
     }
 
-    const auto ctrlPressed = keyEv.IsCtrlPressed();
-    const auto altPressed = keyEv.IsAltPressed();
-    const auto shiftPressed = keyEv.IsShiftPressed();
-
     // Alt key sequences _require_ the char to be in the keyevent. If alt is
     // pressed, manually get the character that's being typed, and put it in the
     // KeyEvent.
     // DON'T manually handle Alt+Space - the system will use this to bring up
     // the system menu for restore, min/maximimize, size, move, close
     wchar_t ch = UNICODE_NULL;
-    if (altPressed && vkey != VK_SPACE)
+    if (states.IsAltPressed() && vkey != VK_SPACE)
     {
         ch = static_cast<wchar_t>(LOWORD(MapVirtualKey(vkey, MAPVK_VK_TO_CHAR)));
         // MapVirtualKey will give us the capitalized version of the char.
         // However, if shift isn't pressed, we want to send the lowercase version.
         // (See GH#637)
-        if (!shiftPressed)
+        if (!states.IsShiftPressed())
         {
             ch = towlower(ch);
         }
     }
 
-    // Manually handle Ctrl+H. Ctrl+H should be handled as Backspace. To do this
-    // correctly, the keyEvents's char needs to be set to Backspace.
-    // 0x48 is the VKEY for 'H', which isn't named
-    if (ctrlPressed && vkey == 0x48)
-    {
-        ch = UNICODE_BACKSPACE;
-    }
-    // Manually handle Ctrl+Space here. The terminalInput translator requires
-    // the char to be set to Space for space handling to work correctly.
-    if (ctrlPressed && vkey == VK_SPACE)
+    if (states.IsCtrlPressed())

```

Comment:
I disliked how the old code had two if-conditions beginning with `if (ctrlPressed ...)`.
The new code is an _identical_ transformation to a switch-case instead.
This change is one I also tried to submit in #1436, but reverted it to simplify the old PR.

Reply:
Excellent.